### PR TITLE
Only initialize program builtins

### LINKED ISSUE #1814: Add CairoRunner:: initialize_program_builtins method
Add a `initialize_program_builtins` method to the CairoRunner struct.
The goal of this method, is to initialize just the `&self.program.builtins`
This function should be the same as the `initialize_all_builtins` but with some modifications:

Remove these chuncks of code:

```
        let starknet_preset_builtins = vec![
            BuiltinName::pedersen,
            BuiltinName::range_check,
            BuiltinName::output,
            BuiltinName::ecdsa,
            BuiltinName::bitwise,
            BuiltinName::ec_op,
            BuiltinName::keccak,
            BuiltinName::poseidon,
        ];
```

```
        for builtin_name in starknet_preset_builtins {
            if !self.program.builtins.contains(&builtin_name) {
                initialize_builtin(builtin_name, &mut self.vm, add_segment_arena_builtin)
            }
        }
```

note: We can avoid the segment arena flag, because it is present in `self.program.builtins`

* Remove the old `initialize_all_builtins` method, is not necessary

### PR DESCRIPTION
**Pull request by JulianGCalderon**
# Only initialize program builtins

closes #1814

## Description

Replace initialize_all_builtins for initialize_program_builtins

It also removes/changes tests that used the former function to make sense with the new function

## Checklist
- [x] Linked to Github Issue

### CODE REVIEW

**Review by pefontana — APPROVED** [review]
Nice one @JulianGCalderon !

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 #### Upcoming Changes
 
+* fix(BREAKING): Only initialize program builtins instead of a preset list
```
Comment (fmoletta) [inline CHANGELOG.md:5]: As this is a breaking change I think we should mention which part of the public api was changed. I would add something like: Replace `CairoRunner` method `initialize_all_builtins` with `initialize_program_builtins`.

Comment (JulianGCalderon) [reply CHANGELOG.md]: Done!

**Review by fmoletta — APPROVED** [review]
🚀

**Pull request merged**